### PR TITLE
Document templates

### PR DESCRIPTION
**Pull request by ulrikstrid**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,9 +6,10 @@
 
 Pesy is a CLI tool to assist project management using `package.json` to automatically configure libraries and executables built
 with Dune. Pesy is a great tool to
+
```
Comment (ulrikstrid) [inline README.md]: These changes are autoformatting, are they fine?

File: README.md [hunk 2]
```diff
@@ -559,6 +557,25 @@ esy pesy
 esy build
 ```
 
+### Templates (experimental)
```
Comment (ulrikstrid) [inline README.md]: This is the important part

File: README.md [hunk 3]
```diff
@@ -559,6 +557,25 @@ esy pesy
...
+
+It is possible to create templates to be used by pesy. This is a experimental feature with potential breaking changes without notice.
+
+It works by downloading a git repo and then replacing special strings in filenames and files inside the repo. The special strings are currently these:
+
+| In filename                      | In contents                | Replaced with              |
+| -------------------------------- | -------------------------- | -------------------------- |
+| \_\_PACKAGE_NAME\_\_             | <PACKAGE_NAME>             | package_name_kebab         |
```
Comment (ulrikstrid) [inline README.md]: Should this line and the next really be the same or is this a bug in the template?

File: README.md [hunk 4]
```diff
@@ -559,6 +557,25 @@ esy pesy
 esy build
 ```
 
+### Templates (experimental)
+
+It is possible to create templates to be used by pesy. This is a experimental feature with potential breaking changes without notice.
```
Comment (prometheansacrifice) [inline README.md]:
```suggestion
It is possible to create templates to be used by pesy. This is a experimental feature with potential breaking changes without notice (we need the flexibility at this point in time).
```

File: README.md (hunk 3 above)
Comment (prometheansacrifice) [reply README.md]: I think we missed something, originally PACKAGE_NAME_FULL also contained the scope

Comment (prometheansacrifice) [reply README.md]: Took a closer look. Indeed, the two variables semantically identical. We can get rid of them. At the time of bootstrapping, packages never have scopes.

**Pull request merged**